packages/playwright/tests/test_utils.js

```diff
@@ -0,0 +1,200 @@
+  page.mouse.move(0, 0); // Move away from content on page.
+  await delay(1000);
+
+  // Check whether we are already or still logged in.
+  let navContainer = page.locator('[data-testid=nav-container]');
+  while (await navContainer.isVisible()) {
+    // console.log('Already (still) logged in. Need to logout.');
+    await navContainer.hover({timeout: 5000});
+    const signOutLink = page.locator('[data-testid=sign-out-link]');
+    await expect(signOutLink).toBeVisible();
+
+    await signOutLink.hover({timeout: 5000});
+    await signOutLink.click({timeout: 5000});
+
+    await delay(1000);
+    await page.waitForURL('**/roadmap');
+    await expect(page).toHaveTitle(/Chrome Status/);
+    page.mouse.move(0, 0); // Move away from content on page.
+    // console.log('Should be logged out ow.');
+    await delay(1000);
+
+    navContainer = page.locator('[data-testid=nav-container]');
+  }
+  await delay(1000);
+
+  // Expect login button to be present.
+  // console.info('expect login button to be present and visible');
+  const loginButton = page.locator('button[data-testid=dev-mode-sign-in-button]');
+  await expect(loginButton).toBeVisible({timeout: loginTimeout});
+
+  await loginButton.click({timeout: 1000, delay: 100});
+  await delay(loginTimeout / 3); // longer delay here, to allow for initial login.
+
+  // Expect the title to contain a substring.
+  await expect(page).toHaveTitle(/Chrome Status/);
+  page.mouse.move(0, 0); // Move away from content on page.
+  await delay(6000);
+
+  // Take a screenshot of header that should have "Create feature" button.
+  // console.log('take a screenshot of header that should have "Create feature" button');
+  await expect(page.locator('[data-testid=header]')).toHaveScreenshot('after-login-click.png');
+
+  // Check that we are logged in now.
+  // Expect a nav container to be present.
+  // This sometimes fails, even though the screenshot seems correct.
+  navContainer = page.locator('[data-testid=nav-container]');
+  await expect(navContainer).toBeVisible({timeout: loginTimeout});
+  loginTimeout = 5000; // After first login, reduce timeout/delay.
+
+  // console.log('login: done');
+}
+
+/**
+ * @param {import("playwright-core").Page} page
+ */
+export async function logout(page) {
+  // Attempt to sign out after running each test.
+  // First reset to the roadmap page.
+  // console.log('logout: goto /');
+  await page.goto('/');
+  await page.waitForURL('**/roadmap');
+  await delay(1000);
+
+  await expect(page).toHaveTitle(/Chrome Status/);
+  page.mouse.move(0, 0); // Move away from content on page.
+  await delay(1000);
+
+  const navContainer = page.locator('[data-testid=nav-container]');
+  await expect(navContainer).toBeVisible({timeout: 20000});
+  await delay(1000);
+
+  await navContainer.hover({timeout: 5000});
+  const signOutLink = page.locator('[data-testid=sign-out-link]');
+  await expect(signOutLink).toBeVisible();
+
+  await signOutLink.hover({timeout: 5000});
```

Comment:
Why hover over the sign-out link before clicking it?

Reply:
That part of my experimenting to get it to work reliably.  It might not be needed after all.  I was also taking a screenshot of the hovered link just to see that clicking on it should work.  But there is no screenshot of that now since a failed screenshot during logout made the logout process even more unreliable.  Adding a TODO to check this out further.